lightningd/peer_control.c

```diff
@@ -49,6 +49,8 @@ static void destroy_peer(struct peer *peer)
 	list_del_from(&peer->ld->peers, &peer->list);
 	if (peer->fd >= 0)
 		close(peer->fd);
+	if (peer->gossip_client_fd >= 0)
+		close(peer->gossip_client_fd);
```

Comment:
Yes, I assumed it would.  Of course, this is when we actually *want* to forget all about a peer, not just because it disconnected...